### PR TITLE
(refactor) Modal Transaction Editing

### PR DESCRIPTION
**Pull request by sfount**
This PR implements a proposal for viewing/ editing financial transactions in a modal. This would replace the current Journal editing and provide the option for any module to allow the user to inspect or edit any non-posted transaction. 

*Screenshots and description to follow*

### CODE REVIEW

**Comment by sfount** [comment]
E2E tests will need to be updated for the new method of editing before this PR can be merged.

File: client/src/js/services/grid/GridGrouping.js [hunk 1]
```diff
@@ -109,7 +97,7 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     gridRows.forEach(function (row) {
       var parentRow = row.treeNode.parentRow;
 
-      if (isUnusedParentRow(parentRow)) {
+      if (parentRow && isUnusedParentRow(parentRow)) {
```
Comment (jniles) [inline client/src/js/services/grid/GridGrouping.js:100]: Nice catch! :+1:

File: client/src/js/services/grid/GridGrouping.js [hunk 2]
```diff
@@ -85,18 +85,6 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     var gridApi = this.gridApi;
 
     this.selectedRowCount = gridApi.selection.getSelectedCount();
-
-    // determine that this selection is a header row
```
Comment (jniles) [inline client/src/js/services/grid/GridGrouping.js:89]: :+1:

File: client/src/js/services/grid/GridSorting.js [hunk 3]
```diff
@@ -57,14 +57,14 @@ function GridSortingService(util) {
    */
   function GridSorting(gridOptions) {
```
Comment (jniles) [inline client/src/js/services/grid/GridSorting.js:58]: Do we still need gridOptions here?

File: client/src/modules/journal/journal.html [hunk 4]
```diff
@@ -4,16 +4,11 @@
       <li class="static" translate>TREE.FINANCE</li>
       <li class="title">
         <span translate>POSTING_JOURNAL.TITLE</span>
-
-        <span class="badge badge-primary text-uppercase" ng-show="JournalCtrl.transactions.isEditing()">
-          <i class="fa fa-street-view"></i>
```
Comment (jniles) [inline client/src/modules/journal/journal.html:9]: :dancer:

File: client/src/modules/journal/journal.html [hunk 5]
```diff
@@ -89,51 +95,24 @@
...
   <div class="container-fluid">
-
     <div class="row">
       <div class="col-md-12">
-
         <div
           id="journal-grid"
           class="grid-journal"
-          ng-style="JournalCtrl.filterBarHeight"
+          ng-class="{ 'grid-journal' : !JournalCtrl.unknownTransactionEditState, 'grid-journal-alert' : JournalCtrl.unknownTransactionEditState }"
```
Comment (jniles) [inline client/src/modules/journal/journal.html:110]: Do you need this `grid-journal` class?  I think it is redundant.

File: client/src/modules/journal/journal.js [hunk 6]
```diff
@@ -493,90 +393,70 @@ function JournalController(Journal, Sorting, Grouping,
...
-
-        row.hrEntity = entity.hrEntity;
-        transactions.editCell(row, 'entity_uuid', entity.uuid);
+  startup();
+  
+  vm.editTransactionModal = editTransactionModal;
+  function editTransactionModal() { 
+    // block multiple simultaneous edit 
+    if (selection.selected.groups.length > 1) { 
+      Notify.warn('You have multiple transactions selected. Multiple transaction editing is currently disabled');
```
Comment (jniles) [inline client/src/modules/journal/journal.js:410]: This needs to be translated at some point.

File: client/src/modules/journal/journal.service.js [hunk 7]
```diff
@@ -36,18 +42,19 @@ function JournalService(Api, AppCache, Filters, Periods, Modal) {
     };
 
     return service.$http.post('/journal/'.concat(entity.uuid, '/edit'), saveRequest)
-      .then(service.util.unwrapHttpRequest);
+      .then(service.util.unwrapHttpResponse);
```
Comment (jniles) [inline client/src/modules/journal/journal.service.js:45]: :dancer:

File: gulpfile.js [hunk 8]
```diff
@@ -189,7 +189,7 @@ gulp.task('client-compile-vendor', () =>
 // writes output to style.min.css
 gulp.task('client-compile-css', () =>
   gulp.src(paths.client.css)
-    .pipe(cssnano())
+    .pipe(cssnano({ zindex : false }))
```
Comment (jniles) [inline gulpfile.js:192]: :+1:  We should definitely announce this in a release.

**Review by jniles — COMMENTED** [review]
Just a few remarks, nothing major.

I'd like to have these changes for my Trial Balance branch ... should I merge with the IMA World Health branch and rebase to that, or should I rebase to yours directly?

**Pull request merged**